src/components/test-utils/mocks/MockI18nextProvider.tsx

```
@@ -0,0 +1,23 @@
+import React, { ReactNode } from 'react';
+import type { i18n } from 'i18next';
+
+/**
+ * A mock implementation of I18nextProvider for testing purposes.
+ * Allows verification of i18n instance provision through data attributes.
+ *
+ * @param props - Component properties
+ * @param props.children - Child elements to be wrapped by the provider
```

ESLint

[error] tsdoc-param-tag-with-invalid-name: The @param block should be followed by a valid parameter name: The identifier cannot non-word characters

(tsdoc/syntax)